OpenRA.Game/Map/CellLayer.cs

```diff
@@ -45,6 +45,19 @@ public void CopyValuesFrom(CellLayer<T> anotherLayer)
 			Array.Copy(anotherLayer.entries, entries, entries.Length);
 		}
 
+		public static CellLayer<T> CreateInstance(Func<MPos, T> initialCellValueFactory, Size size, TileShape tileShape)
+		{
+			var cellInfoLayer = new CellLayer<T>(tileShape, size);
```

Comment:
nit: This doesn't necessarily hold cellInfo, so should be renamed to the generic `cellLayer`.